Library/Homebrew/cmd/commands.rb

```diff
@@ -8,51 +8,70 @@ module Homebrew
   module_function
 
   def commands
-    if ARGV.include? "--quiet"
-      cmds = internal_commands + external_commands
-      cmds += internal_developer_commands
-      cmds += HOMEBREW_INTERNAL_COMMAND_ALIASES.keys if ARGV.include? "--include-aliases"
-      puts Formatter.columns(cmds.sort)
-    else
-      # Find commands in Homebrew/cmd
-      puts "Built-in commands"
-      puts Formatter.columns(internal_commands)
-
-      # Find commands in Homebrew/dev-cmd
-      puts
-      puts "Built-in developer commands"
-      puts Formatter.columns(internal_developer_commands)
-
-      # Find commands in the path
-      unless (exts = external_commands).empty?
-        puts
-        puts "External commands"
-        puts Formatter.columns(exts)
+    # TODO: Goal is to not require this function at all.
+    command = CommandsCommand.new
+    command.check_for_errors
+    command.commands
+  end
+
+  class CommandsCommand < Command
+    def initialize
+      options do
```

Comment:
@MikeMcQuaid, I'll try to figure out how to do this while also keeping instance-level implementation of `Command` class. There might be some ruby hacks out there. I'll look for them.

One non-hacky way to achieve this would be to do class level implementation of `Command` functions. (this was actually done in my previous PR: #2420 )